iroh-net/src/magic_endpoint/rtt_actor.rs

```diff
@@ -0,0 +1,153 @@
+//! Actor which coordinates the congestion controller for the magic socket
+
+use std::collections::HashMap;
+
+use futures_concurrency::stream::stream_group;
+use futures_lite::StreamExt;
+use iroh_base::key::NodeId;
+use tokio::sync::{mpsc, Notify};
+use tokio::task::JoinHandle;
+use tokio::time::Duration;
+use tracing::{debug, error, info_span, trace, warn, Instrument};
+
+use crate::magicsock::{ConnectionType, ConnectionTypeStream};
+
+#[derive(Debug)]
+pub(super) struct RttHandle {
+    pub(super) _handle: JoinHandle<()>,
```

Comment:
The actor is terminated by dropping the mpsc sender.  I added the JoinHandle here with the intention of allowing panic propagation.  But then didn't use it because that's very painful to do in something like the `MagicEndpoint`, i.e. an object that is supported by long-running tasks but itself does not have a task. 

Maybe there's a world where it checks on each method called whether any of it's actors have errored or paniced and propagates that error/panic.  And that's why I added a Joinhandle here.  But that change shouldn't be included in this PR.